app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -95,20 +111,35 @@ class UserViewModel(
   }
 
   /**
-   * Updates a user in the Firestore database.
+   * Updates a user in the Firestore database. If a user has Uri for the profile picture, it will
+   * upload the image to the cloud storage and then delete the Uri.
    *
    * @param user the user to update
    */
-  fun updateUser(user: User) {
-    userRepository.updateUser(
-        user,
-        onSuccess = { Log.d("UserViewModel", "User updated") },
-        onFailure = { exception ->
-          Log.e(
-              "com.github.se.cyrcle.model.user.UserViewModel",
-              "Failed to update user: ${user.public.userId}",
-              exception)
-        })
+  fun updateUser(user: User, context: Context? = null) {
```

Comment:
You forgot to document the `context` argument here as well.

Reply:
Solved

Reply:
Solved